sale_coupon_domain_product_discount/report/sale_coupon_report_template.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template id="report_coupon" inherit_id="sale_coupon.report_coupon">
+        <xpath
+            expr="//t[@t-if='len(o.program_id.discount_specific_product_ids) > 1']"
+            position="after"
+        >
+            <t t-elif="o.program_id.discount_apply_on_domain_product">
+                <t t-set="text">OFF ON SOME PRODUCTS*</t>
```

Comment:
Think in the final result. This is how it looks right now. It doesn't make much sense. Use a similar technique to the one that core uses to print the list of products but 

![image](https://user-images.githubusercontent.com/5040182/222378554-7b2effa8-fc96-4503-9223-e5e80af27d52.png)

Reply:
corrected, thanks!